### ISSUE TITLE
Docker image

### ISSUE THREAD

**Issue by Ailothaen** [Ailothaen/RedditArchiver#1]
Some people wanted to be able to run the RedditArchiver app in a Docker container.

Despite I took some Docker classes some time ago, I do not have a lot of experience with it, and there are some aspects of RedditArchiver that makes making a Dockerfile complicated for me (or more exactly: makes me doubtful about whether the way I do is the correct one)

In particular, there are two aspects of RedditArchiver that I do not really know how to handle:

- RedditArchiver has a optional feature to block requests coming from non-listed IPs. How to ensure the requesting IP is passed correctly to the container? (Might be difficult in complex orchestrator setups)
- How to handle persistent/variable data (logs, config files, output HTML files...)? What should be used? (Volumes, mappings... ?)

Here is the uncomplete Dockerfile I came up with, as now:

```Dockerfile
FROM python

WORKDIR /opt/redditarchiver
ENV RA_DOCKER=1
COPY src .

RUN pip install -r requirements.txt

EXPOSE 80:80
CMD ["gunicorn", "wsgi:app", "-b", "0.0.0.0:80", "-w", "1"]
```

Any help or suggestions are welcome.

**Comment by dan1elt0m** [comment]
Hey @Ailothaen, see PR https://github.com/Ailothaen/RedditArchiver/pull/3

Regarding your concern about the allowed-ips. The ip to add is the docker internal local host ip. During docker build this docker ip  is automatically added to the config.yml. The docker localhost ip is configurable in the compose.yml. I have set it to 192.168.65.0/24, which is the default ip.

Regarding persistent data concern: I added two volumes to persist logs and output data. Let me know if there is something else you would like to persist.

**Comment by Ailothaen** [comment]
Hello,

I pushed recently a commit (fa8e2a1223c04e8aa483f69240525ecdd9c23164) that added a Dockerfile and made the several changes that were necessary in the code. I eventually went with a simpler approach, in order to disrupt the codebase the least as possible.

So RedditArchiver can be now ran with Docker. I close this issue as I consider it resolved, but another one may be opened in the future, if a Docker-related feature request comes up.

Thank you again for your help and your advices!
